File: src/modules/algorithm/seed/algorithms/queue/go/array-solution.go
package queue

// Queue represents a queue data structure implemented using an array.
type Queue struct {
	items    []interface{}
	capacity int
	front    int
	rear     int
	count    int
}

// NewQueue creates a new queue with the given capacity.
func NewQueue(capacity int) *Queue {
	return &Queue{
		items:    make([]interface{}, capacity),
		capacity: capacity,
		front:    0,
		rear:     -1,
		count:    0,
	}
}

// Enqueue adds an element to the rear of the queue.
func (q *Queue) Enqueue(item interface{}) {
	if q.IsFull() {
		panic("Queue Overflow")
	}
	q.rear = (q.rear + 1) % q.capacity
	q.items[q.rear] = item
	q.count++
}

// Dequeue removes an element from the front of the queue.
func (q *Queue) Dequeue() interface{} {
	if q.IsEmpty() {
		return nil
	}
	item := q.items[q.front]
	q.front = (q.front + 1) % q.capacity
	q.count--
	return item
}

// Peek returns the element at the front of the queue without removing it.
func (q *Queue) Peek() interface{} {
	if q.IsEmpty() {
		return nil
	}
	return q.items[q.front]
}

// IsEmpty checks if the queue is empty.
func (q *Queue) IsEmpty() bool {
	return q.count == 0
}

// IsFull checks if the queue is full.
func (q *Queue) IsFull() bool {
	return q.count == q.capacity
}

// Size returns the number of elements in the queue.
func (q *Queue) Size() int {
	return q.count
} 

File: src/modules/algorithm/seed/algorithms/queue/python/array-solution.py
class Queue:
    def __init__(self, capacity: int):
        """
        Initializes an empty queue with a given capacity.
        """
        self.items = [None] * capacity
        self.capacity = capacity
        self.front = 0
        self.rear = -1
        self.count = 0

    def enqueue(self, item: any) -> None:
        """
        Adds an element to the rear of the queue.
        """
        if self.is_full():
            raise OverflowError("Queue is full")
        self.rear = (self.rear + 1) % self.capacity
        self.items[self.rear] = item
        self.count += 1

    def dequeue(self) -> any:
        """
        Removes an element from the front of the queue.
        """
        if self.is_empty():
            return None
        item = self.items[self.front]
        self.front = (self.front + 1) % self.capacity
        self.count -= 1
        return item

    def peek(self) -> any:
        """
        Returns the element at the front of the queue without removing it.
        """
        if self.is_empty():
            return None
        return self.items[self.front]

    def is_empty(self) -> bool:
        """
        Checks if the queue is empty.
        """
        return self.count == 0

    def is_full(self) -> bool:
        """
        Checks if the queue is full.
        """
        return self.count == self.capacity

    def size(self) -> int:
        """
        Returns the number of elements in the queue.
        """
        return self.count 